### ISSUE TITLE
Add "Ignore installed package" option

### ISSUE THREAD

**Issue by SiJiL82** [microsoft/winget-cli#1439]
There are several packages that have been added by the community that are problematic / just incorrect, which prevents you from doing an "upgrade all".
It would be very useful to have an "ignore" option where you can specify an installed program to just exclude from the winget upgrade functionality.

For example:

Name                                            Id                           Version      Available    Source
-------------------------------------------------------------------------------------------------------------
Unity                                           UnityTechnologies.Unity      2020.3.16f1  2021.1.19f1  winget

Unity 2021 is not an upgrade from 2020 - it is a completely separate application, and the manifest for it should not have been created like this. Rather than have it permanently there every time I do an upgrade search, it would be much better to be able to just never see it.

Obviously it would be better to not have these problematic packages in the first place, but there doesn't currently seem to be an effective way to get them removed / fixed.

**Comment by siddagra** [comment]
Agreed. I proposed a similar issue but I think a better way to fix this may be to just state the errors and move on to upgrading the next package instead of failing the entire upgrade sequence. I mention this here: https://github.com/microsoft/winget-cli/issues/1442

**Comment by jedieaston** [comment]
We've had the discussion about Unity before. The problem is that we want to have multiple versions of Unity in the repo, so that people can just install what they want via `--version`, but then `winget upgrade` makes a mess of things.

#476 is the standing solution to the problem, and I guess it's on the backlog for the next major release after 1.1 (I have absolutely no idea when that is though, since 1.1 hasn't had any public builds yet).

**Comment by saxc** [comment]
I would also prefer an ignore list. I have several cases in which I want to use this.

Here are two examples.

- The GOG GALAXY installed version shows me as Unknown. One each upgrade --all it downloads the file and starts the installer, the installer shows the message that this or a newer version is installed.

- By an Visual Studio Update i prefer the Visual Studio installer to install the Update.

**Comment by denelon** [comment]
@jedieaston the release yesterday https://github.com/microsoft/winget-cli/releases/tag/v1.0.12576 has most of the work for Windows Package Manager v1.1 we're doing some additional internal testing and a few more tweaks. It's very close [tm].

**Comment by jdhitsolutions** [comment]
This idea of ignoring packages isn't always tied to a repo. This is one example. 
![image](https://user-images.githubusercontent.com/12089920/143957431-0067e555-adf9-465e-87a4-7bc34b535c88.png)
I manually installed this app, yet winget detects it. There really is no upgrade. v1.5.1 is it.  It would be helpful that have a way to suppress this item.

**Comment by SiJiL82** [comment]
The ideal solution to that though is to get in contact with the developer and ask them to update their installer so it reports the version correctly. Then it'll work with winget as expected.

**Comment by jdhitsolutions** [comment]
The problem is that this is a legacy app. Unfortunately, Winget detects and reports on all applications, even those installed manually. So either Winget needs a way to only report on apps *installed* with Winget or a suppression mechanism.

**Comment by jedieaston** [comment]
Thoughts on https://github.com/microsoft/winget-cli/issues/1692#issuecomment-981165560 as a solution?

**Comment by biswasmoheshwar** [comment]
I also need this,

**Comment by magic280z** [comment]
Add to this something like Citrix Workspace is wrong 19.12.3000.6 is LTSR it should upgrade to 19.12.6000.x not 22.2.0.4525.  Looks like the same thing for Office 365.  Update channel needs to be supported.  Until then exclusions and manually handling of excluded items better.

                            Citrix.Workspace                  19.12.3000.6     22.2.0.4525      winget

**Comment by chbwien** [comment]
E.g. Irfanview does everything correct. But companies get special versions with the same version number. Winget does not see the difference and will upgrade with the public version.... So there are reasons to suppress upgrades for some packages. This also might be a license restriction (e.g. updates are only within the license for one year, every update after that breaks the app)!

**Comment by denelon** [comment]
@chbwien you should probably also take a look at:
* #476 

as well.

I wasn't aware of this specific scenario. I would hope there would be some metadata differences that would allow us to distinguish between these different versions of a package, but now I understand. Thanks so much for providing the explicit scenario here. Do you know of any specific applications that exhibit this behavior? I'd love to have a test case.
